akka-http-core/src/main/scala/akka/http/impl/settings/ConnectionPoolSettingsImpl.scala

```diff
@@ -33,6 +36,7 @@ object ConnectionPoolSettingsImpl extends SettingsCompanion[ConnectionPoolSettin
   def fromSubConfig(root: Config, c: Config) = {
     ConnectionPoolSettingsImpl(
       c getInt "max-connections",
```

Comment:
It will be valid, the Settings classes are specifically designed such that we can safely add settings (uff that was painful back then but now we reap the benefits :-)).

You'll need to add:

```
[info] akka-http-core: found 2 potential binary incompatibilities while checking against com.typesafe.akka:akka-http-core_2.11:2.4.2  (filtered 126)
[error]  * abstract method getMinConnections()Int in class akka.http.javadsl.settings.ConnectionPoolSettings is present only in current version
[error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.http.javadsl.settings.ConnectionPoolSettings.getMinConnections")
[error]  * abstract method minConnections()Int in class akka.http.scaladsl.settings.ConnectionPoolSettings is present only in current version
[error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.http.scaladsl.settings.ConnectionPoolSettings.minConnections")
```

to `MiMa.scala` (this output is in jenkins, once you click the expand all of the log link and search for " incompatibilities "